landoapi/transplants.py

```diff
@@ -226,6 +227,16 @@ def revision_check_pairs(self) -> list[tuple[dict, dict]]:
             for revision in self.stack_data.revisions.values()
         ]
 
+    def get_repo_for_revision(self, revision: dict) -> Optional[Repo]:
+        """Given a revision object, return the associated `Repo` in Lando."""
+        repo_phid = PhabricatorClient.expect(revision, "fields", "repositoryPHID")
+
+        repo = self.stack_data.repositories.get(repo_phid)
+        if not repo or not self.supported_repos:
+            return None
+
+        return self.supported_repos.get(repo["fields"]["shortName"])
```

Comment:
Any reason this isn't `self.supported_repos[repo["fields"]["shortName"]]`? Could fail sooner here if there is something wrong or missing.

Reply:
It's possible to have repos on Phabricator that aren't supported in Lando, and in that case we want to just return `None` and not fail.